coordinator/statement_executor.go

```diff
@@ -394,6 +385,12 @@ func (e *StatementExecutor) executeSetPasswordUserStatement(q *influxql.SetPassw
 }
 
 func (e *StatementExecutor) executeSelectStatement(stmt *influxql.SelectStatement, ctx *influxql.ExecutionContext) error {
+	// If there is a target and we are in a read only context, exit early since
+	// we will return an error.
+	if stmt.Target != nil && ctx.ReadOnly {
```

Comment:
This looks like a breaking change.  If it is, I'd rather we leave it as is for backwards compatibility since many clients would need to be updated.